### PR TITLE
Digital Shadows - bug fix in fetch incidents

### PR DESCRIPTION
**Pull request by dantavori**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/18362

## Description
Fixed an issue where ***fetch-incidents*** command returned duplicated incidents.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/DigitalShadows/Integrations/integration-DigitalShadows_CHANGELOG.md)

### CODE REVIEW

**Comment by dantavori** [comment]
will add a TPB

File: Integrations/integration-DigitalShadows.yml [hunk 1]
```diff
@@ -23,8 +23,8 @@ configuration:
   defaultvalue: ""
   type: 4
   required: true
-- display: Trust any certificate (unsecure)
-  name: insecure
+- display: Trust any certificate (not secure)
+  name: inse
```
Comment (dantavori) [inline Integrations/integration-DigitalShadows.yml]:
```suggestion
  name: insecure
```

File: Integrations/integration-DigitalShadows.yml [hunk 2]
```diff
@@ -755,14 +755,23 @@ script:
             parsed_res.content.forEach(function(inc){
                 inc['origin'] = name;
                 incidents.push({
-                  name: name, 
+                  name: name,
```
Comment (dantavori) [inline Integrations/integration-DigitalShadows.yml]:
```suggestion
                  name: name, 
```

File: Integrations/integration-DigitalShadows.yml [hunk 3]
```diff
@@ -771,14 +780,30 @@ script:
...
+
+        // second API call - fetch all incidents (within the date range)
         bodyArgs = {filter_dateRange: lastRun.value + '/' + now};
-        incidents = incidents.concat(fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'Client Incident'));
+        temp = fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'Client Incident');
+
+        // remove duplicated incidents from second API call that resulted in the first call
+        if (incidents) {
+            incidents = incidents.concat(temp.filter(function(inc) {
+               return ids.indexOf(JSON.parse(inc.rawJSON).id) == -1;
```
Comment (guykeller) [inline Integrations/integration-DigitalShadows.yml]: Let's split this into several lines with clear names for readability

File: Integrations/integration-DigitalShadows.yml [hunk 4]
```diff
@@ -771,14 +780,30 @@ script:
...
         var incidents = fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'PHISHING_ATTEMPT');
 
+        var ids = getIncidentsIds(incidents);
+
+        // second API call - fetch all incidents (within the date range)
         bodyArgs = {filter_dateRange: lastRun.value + '/' + now};
-        incidents = incidents.concat(fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'Client Incident'));
+        temp = fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'Client Incident');
+
+        // remove duplicated incidents from second API call that resulted in the first call
```
Comment (guykeller) [inline Integrations/integration-DigitalShadows.yml]: let's try to improve this comment

File: Integrations/integration-DigitalShadows.yml [hunk 5]
```diff
@@ -771,14 +780,30 @@ script:
...
+        if (incidents) {
+            incidents = incidents.concat(temp.filter(function(inc) {
+               return ids.indexOf(JSON.parse(inc.rawJSON).id) == -1;
+            }));
+        }
+        else {
```
Comment (guykeller) [inline Integrations/integration-DigitalShadows.yml]: Why is the 'else' case needed?

Comment (dantavori) [reply Integrations/integration-DigitalShadows.yml]: Without separating the cases to if-else, `ids` would be a NoneType object and there will be a null pointer exception in line 796.

File: Integrations/integration-DigitalShadows.yml (hunk 3 above)
Comment (dantavori) [reply Integrations/integration-DigitalShadows.yml]:
```suggestion
               incidentId = JSON.parse(inc.rawJSON).id;
               notDuplicatedIncident = (ids.indexOf(incidentId) == -1);
               return notDuplicatedIncident;
```

File: Integrations/integration-DigitalShadows.yml (hunk 4 above)
Comment (dantavori) [reply Integrations/integration-DigitalShadows.yml]:
```suggestion
        // concatenate only new incident to list (remove duplicates)
```

**Pull request merged**